src/main/java/commission/dao/SaleDaoImpl.java

```diff
@@ -6,75 +6,66 @@
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
 
+import javax.sql.DataSource;
 import java.util.List;
 import java.util.Optional;
 
 @Repository
 public class SaleDaoImpl implements SaleDao{
 
-    JdbcTemplate jdbcTemplate;
+    private final JdbcTemplate jdbcTemplate;
+    private final SalesRowMapper rowMapper = new SalesRowMapper();
 
     @Autowired
-    public SaleDaoImpl(JdbcTemplate jdbcTemplate) {
-        this.jdbcTemplate = jdbcTemplate;
+    public SaleDaoImpl(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
     }
 
+
     @Override
     public List<Sale> findAll() {
         var query = "SELECT * FROM sales";
-        return jdbcTemplate.query(query, new SalesRowMapper());
+        return this.jdbcTemplate.query(query, this.rowMapper);
     }
 
 
     @Override
     public void insertSale(Sale sale) {
         var query = "INSERT INTO sales(price, percentage, commission) VALUES (?,?,?)";
-        jdbcTemplate.update(query, sale.price(), sale.percentage(), commissionPerSale(sale));
+        this.jdbcTemplate.update(query, sale.price(), sale.percentage(), sale.commission());
     }
 
     @Override
     public Optional<Sale> findSaleById(long id) {
         var query = "SELECT * FROM sales WHERE id=?";
-        return Optional.ofNullable(jdbcTemplate.queryForObject(query, new SalesRowMapper(), id));
-    }
-
-    @Override
-    public Sale findById(long id) {
-        var query = "SELECT * FROM sales WHERE id=?";
-        return jdbcTemplate.queryForObject(query, new SalesRowMapper(), id);
+        try {
+            return Optional.ofNullable(this.jdbcTemplate.queryForObject(query, this.rowMapper, id));
+        } catch (Exception e) {
+            return Optional.empty();
```

Comment:
I would advise to rather catch the `EmptyResultDataAccessException` specifically. Otherwise you might end up catching ANY exception like sql syntax errors, database connection errors etc.

If you aren't catching a specific exception the at least log the error so that it doesn't get "swallowed". In this case Spring does return a specific exception you can use but just FYI Logger has an error method that takes the exception and a message e.g. `logger.error("Error reading data", e)`